CodePush.ios.js

```diff
@@ -126,75 +131,129 @@ function checkForUpdate() {
  * releases, and displaying a standard confirmation UI to the end-user
  * when an update is available.
  */
-function sync(options = {}) {  
+function sync(options = {}, syncStatusChangeCallback, downloadProgressCallback) {  
   var syncOptions = {
-    descriptionPrefix: " Description: ",
-    appendReleaseDescription: false,
     
     ignoreFailedUpdates: true,
-    
-    mandatoryContinueButtonLabel: "Continue",
-    mandatoryUpdateMessage: "An update is available that must be installed.",
-    
-    optionalIgnoreButtonLabel: "Ignore",
-    optionalInstallButtonLabel: "Install",
-    optionalUpdateMessage: "An update is available. Would you like to install it?",
-    
+    installMode: CodePush.InstallMode.ON_NEXT_RESTART,
     rollbackTimeout: 0,
-    
-    updateTitle: "Update available",
+    updateDialog: null,
     
     ...options 
   };
-
+  
+  syncStatusChangeCallback = typeof syncStatusChangeCallback == "function"
+    ? syncStatusChangeCallback
+    : function(syncStatus) {
+        switch(syncStatus) {
+          case CodePush.SyncStatus.CHECKING_FOR_UPDATE:
+            log("Checking for update.");
+            break;
+          case CodePush.SyncStatus.DOWNLOADING_PACKAGE:
+            log("Downloading package.");
+            break;
+          case CodePush.SyncStatus.AWAITING_USER_ACTION:
+            log("Awaiting user action.");
+            break;
+          case CodePush.SyncStatus.INSTALLING_UPDATE:
+            log("Installing update.");
+            break;
+          case CodePush.SyncStatus.UP_TO_DATE:
+            log("App is up to date.");
+            break;
+          case CodePush.SyncStatus.UPDATE_IGNORED:
+            log("User cancelled the update.");
+            break;
+          case CodePush.SyncStatus.UPDATE_INSTALLED:
```

Comment:
Okay, based on our offline discussion I have changed the behavior to return the UPDATE_INSTALLED status right before the restart. Took me a while because some of my tests were failing. Updated the docs as well.